crates/wit-parser/src/ast/resolve.rs

```diff
@@ -839,6 +842,56 @@ impl<'a> Resolver<'a> {
                     }
                 }
                 ast::InterfaceItem::TypeDef(_) => {}
+                ast::InterfaceItem::Nest(n) => {
+                    if let ast::Nest {
+                        from:
+                            UsePath::Package {
+                                id: package_id,
+                                name,
+                            },
+                        attributes,
+                        docs,
+                    } = n
+                    {
+                        let nest = self
+                            .foreign_deps
+                            .get(&PackageName {
+                                namespace: package_id.namespace.name.to_string(),
+                                name: package_id.name.name.to_string(),
+                                version: package_id.clone().version.map(|v| v.1),
+                            })
+                            .unwrap()
+                            .get(&name.name)
+                            .unwrap();
+                        let nested_id = if let AstItem::Interface(id) = nest {
+                            id
+                        } else {
+                            bail!("Expected interface item")
+                        };
```

Comment:
Could this use `resolve_ast_item_path` and `extract_iface_from_item` to handle fallibility and errors?

Also could you be sure to add a test where `nest` is used with a `world` to ensure it returns an error?